src/Framework/Framework/Compilation/ViewCompiler/ViewCompilingVisitor.cs

```diff
@@ -19,48 +16,44 @@ namespace DotVVM.Framework.Compilation
     public class ViewCompilingVisitor : ResolvedControlTreeVisitor
     {
         protected readonly DefaultViewCompilerCodeEmitter emitter;
-        private readonly CompiledAssemblyCache compiledAssemblyCache;
         protected readonly IBindingCompiler bindingCompiler;
 
         protected int currentTemplateIndex;
-        protected string className;
         protected ControlResolverMetadata? lastMetadata;
         protected string? controlName;
 
-        public ViewCompilingVisitor(DefaultViewCompilerCodeEmitter emitter, CompiledAssemblyCache compiledAssemblyCache, IBindingCompiler bindingCompiler,
-            string className)
+        public Delegate CompiledViewDelegate { get; set; }
+
+        public ViewCompilingVisitor(DefaultViewCompilerCodeEmitter emitter, IBindingCompiler bindingCompiler)
         {
             this.emitter = emitter;
-            this.compiledAssemblyCache = compiledAssemblyCache;
-            this.className = className;
             this.bindingCompiler = bindingCompiler;
+
+            CompiledViewDelegate = new Action(() => new InvalidOperationException("View is not yet compiled."));
         }
 
         public override void VisitView(ResolvedTreeRoot view)
         {
             lastMetadata = view.Metadata;
 
-            var createsCustomDerivedType = view.Metadata.Type == typeof(DotvvmView);
+            var isPageView = view.Metadata.Type == typeof(DotvvmView);
 
-            if (createsCustomDerivedType)
+            if (isPageView)
             {
-                var resultControlType = className + "Control";
-                emitter.ResultControlTypeSyntax = SyntaxFactory.ParseTypeName(resultControlType);
-                emitter.EmitControlClass(view.Metadata.Type, resultControlType);
+                //TODO: Possibly emit custom generated type here
```

Comment:
nah, I don't think that's good for anything

Reply:
Removed